react/src/components/modal.tsx

```diff
@@ -0,0 +1,71 @@
+import React, { useState } from "react";
+import { Button, Form, Input, Modal } from "antd";
+
+import "./modal.css";
+
+interface CustomModalProps {
+  onOk: () => void;
+}
+
+const CustomModal: React.FC<CustomModalProps> = ({ onOk }) => {
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const [form] = Form.useForm();
+
+  const showModal = () => {
+    setIsModalOpen(true);
+  };
+
+  const handleCancel = () => {
+    setIsModalOpen(false);
+  };
+
+  const handleOk = () => {
+    form.validateFields().then(() => {
+      onOk();
+      setIsModalOpen(false);
+      form.resetFields();
+    });
+  };
+
+  return (
+    <>
```

Comment:
Do a `<div className='CreateButton'>` instead of this fragment, then prefix all the classes in the css file with .CreateButton

This is a strategy we do to avoid that the styles overwrite some other component unintentionally